jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/AbstractAdvanceRuntimeDataServiceImpl.java

```diff
@@ -55,58 +58,155 @@ public void setEmf(EntityManagerFactory emf) {
-
-        Query query = entityManager.createNativeQuery(procSQLString);
-        variables.stream().forEach(var -> query.setParameter("NAME_" + var.getColumn(), varPrefix + var.getColumn()));
-        variables.stream().filter(e -> e.getObjectValue() != null).forEach(var -> query.setParameter("VALUE_" + var.getColumn(), var.getObjectValue()));
-        attributes.stream().filter(e -> e.getObjectValue() != null).forEach((entry) -> query.setParameter("ATTR_" + entry.getColumn(), entry.getObjectValue()));
-
-        if (!variables.isEmpty()) {
-            query.setParameter("NUMBER_OF_VARS", variables.size());
+        if (!owners.isEmpty()) {
+            derivedTables.append("INNER JOIN ( \n" +
+                             "           SELECT DISTINCT po.task_id \n" +
+                             "           FROM PeopleAssignments_PotOwners po \n" +
+                             "           WHERE po.entity_id IN (:owners) \n" +
+                             "           GROUP BY po.task_id \n" +
+                             "           HAVING COUNT(po.entity_id) = :num_owners \n" +
+                                 ") pot ON pot.task_id = task.id ");
         }
-        query.setParameter("processType", processType);
 
-        addPagination(query, queryContext);
-        List<Number> ids = query.getResultList();
-        if (ids.isEmpty()) {
-            return Collections.emptyList();
-        }
+        attributes.stream().forEach((expr) -> globalWhere.append(" AND " + computeExpression(expr, expr.getColumn(), ":ATTR_" + expr.getColumn())));
+
+        String procSQLString = mainSQLproducer.apply(derivedTables, globalWhere);
 
-        // now we get the information
+        List<Number> ids = emptyList();
+        EntityManager entityManager = emf.createEntityManager();
+        try {
+            Query query = entityManager.createNativeQuery(procSQLString);
+            taskVariables.stream().forEach(var -> query.setParameter("V_NAME_" + var.getColumn(), var.getColumn()));
+            taskVariables.stream().filter(e -> e.getObjectValue() != null).forEach((var) -> query.setParameter("V_VALUE_" + var.getColumn(), var.getObjectValue()));
+
+            if (!taskVariables.isEmpty()) {
+                query.setParameter("NUMBER_OF_TASKVARS", taskVariables.stream().map(QueryParam::getColumn).distinct().count());
+            }
 
+            processVariables.stream().forEach(var -> query.setParameter("P_NAME_" + var.getColumn(), varPrefix + var.getColumn()));
+            processVariables.stream().filter(e -> e.getObjectValue() != null).forEach(var -> query.setParameter("P_VALUE_" + var.getColumn(), var.getObjectValue()));
+
+            if (!processVariables.isEmpty()) {
+                query.setParameter("NUMBER_OF_PROCVARS", processVariables.stream().map(QueryParam::getColumn).distinct().count());
+            }
+
+            if (!owners.isEmpty()) {
+                query.setParameter("num_owners", owners.size());
```

Comment:
what if `List<String> owners` contains a duplicate owner? the query won't return any data as per query                              `HAVING COUNT(po.entity_id) = :num_owners`

```suggestion
                query.setParameter("num_owners", owners.stream().distinct().collect(Collectors.toList()));
```